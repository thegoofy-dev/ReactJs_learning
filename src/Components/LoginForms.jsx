export function LoginForm() {
    return (
    <form onSubmit={(e)=> {
        e.preventDefault();
        const formdata=  new FormData(e.target);
        const username = formdata.get("username");
        const password = formdata.get("password");

        fetch("http://localhost:5173/api/login", {
            body: {
                username,
                password
            },
            method: "POST",
        })
        }}>
        <label htmlFor="username">UserName:</label>
        <input id='username' onChange={(e) => {
            console.log("UserName:",e.target.value);
        }} 
        name='username'/>
        <br />
        <label htmlFor="password">Password:</label>
        <input id='password' type="password" onChange={(e) => {
            console.log("Password:",e.target.value);
        }}
        name="password"/>
        <br />
        <button >Login</button>
    </form>
    )
}