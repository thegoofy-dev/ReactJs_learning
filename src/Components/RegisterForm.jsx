import { useState } from "react";

export function RegisterForm() {
  // const [username, setUsername] = useState("");
  // const [password, setPassword] = useState("");
  // const [displayName, setDisplayName] = useState("");

  // const isDisabled = !username || !password || !displayName;

  const [formFields, setFormFields] = useState({
    username: "",
    password: "",
    displayName: "",
  });

  return (
    <form>
      <div>
        <label htmlFor="username">UserName:</label>
        <input
          id="username"
          value={formFields.username}
          onChange={(e) => {
            setFormFields((currentSate) => ({
              ...currentSate,
              username: e.target.value,
            }));
          }}
        />
      </div>
      <br />
      <div>
        <label htmlFor="password">Password:</label>
        <input
          id="password"
          type="password"
          value={formFields.password}
          onChange={(e) => {
            setFormFields((currentSate) => ({
              ...currentSate,
              password: e.target.value,
            }));
          }}
        />
      </div>
      <br />
      <div>
        <label htmlFor="displayName">DisplayName:</label>
        <input
          id="displayName"
          value={formFields.displayName}
          onChange={(e) => {
            setFormFields((currentSate) => ({
              ...currentSate,
              displayName: e.target.value,
            }));
          }}
        />
      </div>
      {/* <button disabled={isDisabled}>Login</button> */}

      <div>
        <h3>UserName:</h3> <span>{formFields.username}</span>
        <h3>Password:</h3> <span>{formFields.password}</span>
        <h3>DisplayName:</h3> <span>{formFields.displayName}</span>
      </div>
    </form>
  );
}
