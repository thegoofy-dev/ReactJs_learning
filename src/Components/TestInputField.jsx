export function TestInputField() {
    return <>
      <input type="text" placeholder="Enter data..." value="hello" />
      </>
}