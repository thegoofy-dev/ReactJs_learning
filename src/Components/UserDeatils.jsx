import PropTypes from "prop-types";
import { useState } from "react";

export function UserDetails({ user, setUsers }) {
  const [isEditing, setIsEditing] = useState(false);
  const [username, setUsername] = useState(user.username);
  const [email, setEmail] = useState(user.email);

  const handleSave = () => {
    setUsers((currentUserState) =>
      currentUserState.map((currentUser) =>
        currentUser.id === user.id
          ? { ...currentUser, username, email }
          : currentUser
      )
    );
    setIsEditing(false);
  };

  const handleDelete = () => {
    setUsers((currentUserState) =>
      currentUserState.filter((currentUser) => currentUser.id !== user.id)
    );
  };

  return (
    <div>
      <b>ID:</b> <span>{user.id}</span> <br />
      {isEditing ? (
        <label htmlFor="username">Username: </label>
      ) : (
        <b>Username: </b>
      )}
      {isEditing ? (
        <input
          aria-label="username"
          id="username"
          name="username"
          value={username}
          onChange={(e) => setUsername(e.target.value)}
        />
      ) : (
        <span>{user.username}</span>
      )}
      <br />
      {isEditing ? <label htmlFor="email">Email: </label> : <b>Email: </b>}
      {isEditing ? (
        <input
          aria-label="email"
          id="email"
          type="email"
          name="email"
          value={email}
          onChange={(e) => setEmail(e.target.value)}
        />
      ) : (
        <span>{user.email}</span>
      )}
      <br />
      <div>
        <button data-testid={`edit-btn-${user.id}`} onClick={() => setIsEditing((prev) => !prev)}>
          {isEditing ? "Cancel" : "Edit"}
        </button>
        <button data-testid={`delete-btn-${user.id}`} onClick={handleDelete}>Delete</button>
        {isEditing && <button onClick={handleSave}>Save</button>}
      </div>
      <hr />
    </div>
  );
}

UserDetails.propTypes = {
  user: PropTypes.shape({
    id: PropTypes.number.isRequired,
    username: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
  }).isRequired,
  setUsers: PropTypes.func.isRequired,
};
