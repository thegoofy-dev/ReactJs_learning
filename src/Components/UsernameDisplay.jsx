export function UsernameDisplay({username}) {
    return (
        <div>
            <h2>{username}</h2>
        </div>
    )
}
