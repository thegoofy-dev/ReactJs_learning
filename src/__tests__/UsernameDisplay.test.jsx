import { describe, it, expect } from "vitest"; 
import { render } from "@testing-library/react";
import { UsernameDisplay } from "../Components/UsernameDisplay";

describe("UserName Display", () => {
    it("should render username", () => {
        const result = render(<UsernameDisplay username={"Tillu"} />);
        expect(result.container).toMatchSnapshot();
    });
})