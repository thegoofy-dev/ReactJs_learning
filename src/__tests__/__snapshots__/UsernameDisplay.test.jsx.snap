// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserName Display > should render username 1`] = `
<div>
  <div>
    <span>
      Tillu
    </span>
  </div>
</div>
`;
